addon/ext.js

```diff
@@ -47,8 +47,15 @@ Store.reopen({
   createFragment: function(modelName, props) {
     Ember.assert("The '" + modelName + "' model must be a subclass of MF.Fragment", this.isFragment(modelName));
 
-    var type = this.modelFor(modelName);
-    var internalModel = new InternalModel(type, null, this, getOwner(this).container);
+    var internalModel;
+    if (InternalModelPrototype.hasOwnProperty('modelClass')) {
```

Comment:
I'm not sure if it is an issue in v8 anymore but looks like hasOwnProperty has caused some [slowdown in the past](https://github.com/emberjs/data/pull/4313) with ember. Since InternalModelPrototype perhaps we could set the result of this check in a variable outside the function and reference that?